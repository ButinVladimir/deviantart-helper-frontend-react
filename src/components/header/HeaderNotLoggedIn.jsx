import React from 'react';
import PropTypes from 'prop-types';
import Navbar from 'react-bulma-components/lib/components/navbar';
import Button from 'react-bulma-components/lib/components/button';
import * as serverRoutes from '../../consts/server-routes';
import Config from '../../config/config';

export default function HeaderNotLoggedIn({
  userInfoLoading,
  config,
}) {
  const loginUrl = `${config.serverUrl}${serverRoutes.SERVER_ROUTE_AUTH_CONNECT}`;

  return (
    <Navbar.Brand>
      <Navbar.Item>
        <Button id="login-button" color="primary" loading={userInfoLoading} onClick={() => { window.location = loginUrl; }}>Log in</Button>
      </Navbar.Item>
    </Navbar.Brand>
  );
}

HeaderNotLoggedIn.propTypes = {
  userInfoLoading: PropTypes.bool.isRequired,
  config: PropTypes.instanceOf(Config).isRequired,
};
