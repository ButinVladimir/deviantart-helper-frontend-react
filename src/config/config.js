/**
 * Config.
 */
export default class Config {
  /**
   * @description
   * The constructor.
   */
  constructor() {
    this.serverUrl = process.env.REACT_APP_SERVER_URL;
  }
}
