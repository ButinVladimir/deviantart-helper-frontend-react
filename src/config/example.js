import Config from './config';

const config = new Config(
  'http://localhost:8000', // Server URL.
);

export default config;
