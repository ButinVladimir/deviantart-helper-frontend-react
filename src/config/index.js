import config from './development';

export default config;
