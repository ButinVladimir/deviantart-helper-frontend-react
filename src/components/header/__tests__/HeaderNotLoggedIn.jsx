import React from 'react';
import { shallow } from 'enzyme';
import HeaderNotLoggedIn from '../HeaderNotLoggedIn';
import config from '../../../config';

describe('HeaderNotLoggedIn', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <HeaderNotLoggedIn
        userInfoLoading
        config={config}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
