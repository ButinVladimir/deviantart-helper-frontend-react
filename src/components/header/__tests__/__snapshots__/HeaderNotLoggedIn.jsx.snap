// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNotLoggedIn renders correctly 1`] = `
<NavbarBrand
  className=""
  style={Object {}}
>
  <NavbarItem
    active={false}
    className=""
    dropdown={false}
    dropdownUp={false}
    hoverable={false}
    renderAs="a"
    style={Object {}}
  >
    <Button
      className=""
      color="primary"
      disabled={false}
      fullwidth={false}
      inverted={false}
      isSelected={false}
      isStatic={false}
      loading={true}
      onClick={[Function]}
      outlined={false}
      remove={false}
      renderAs="button"
      reset={false}
      rounded={false}
      size={null}
      state={null}
      style={Object {}}
      submit={false}
      text={false}
    >
      Log in
    </Button>
  </NavbarItem>
</NavbarBrand>
`;
